Added description of queue usage in Smart Power (#120)
Volue-Public/energy-smp-docs: docs/operations/queue-usage-in-smart-power.md

# Queue usage in Smart Power

> In Smart Power message queues are the standard technology to use in asynchronous integrations with both internal and external applications. This is integrations that previously would have been implemented using file based exchange of information.

Reviewer: @torsannes In general, we should avoid using concepts like "previous" or "current", as they are relative to "now". There could be people reading this in 2 or 5 years, who would not have any idea what "previous" means. Could we be more specific? E.g. "in versions prior to", "in legacy systems", "in SmG"?

Author: Changed to "in SmG"

> In Smart Power message queues are the standard technology to use in asynchronous integrations with both internal and external applications. This is integrations that previously would have been implemented using file based exchange of information.

Reviewer: ```suggestion
In Smart Power message queues are the standard technology to use in asynchronous integrations with both internal and external applications. These are integrations that previously would have been implemented using file based exchange of information.
```

Author: fixed

> ## Current usage in existing SmP applications:

Reviewer: ```suggestion
## Current usage in existing SmP applications
```

Author: fixed

* Imports to and exports from Mesh Data Transfer from and to external applications/systems.

> * Activation plans  and heartbeats from Activation Request to Scada Gateway and Mesh Data Transfer.

Reviewer: ```suggestion
* Activation plans and heartbeats from Activation Request to SCADA Gateway and Mesh Data Transfer.
```

Author: fixed

> * Acknowledgements from Scada Gateway to Activation Request.

Reviewer: ```suggestion
* Acknowledgements from SCADA Gateway to Activation Request.
```

Author: fixed

> * Production plans from Mesh Data Transfer to Scada Gateway.

Reviewer: ```suggestion
* Production plans from Mesh Data Transfer to SCADA Gateway.
```

Author: fixed

> ## Queue definitions per Smart Power application:

Reviewer: ```suggestion
## Queue definitions per Smart Power application
```

Author: fixed

> * **Mesh Data Transfer** to/from exxternal applications/systems:

Reviewer: ```suggestion
* **Mesh Data Transfer** to/from external applications/systems:
```

Author: fixed

>   * **ImportQueue** - contains messages to be imported. There may be used several queues with different priorities.

Reviewer: ```suggestion
  * **ImportQueue** - contains messages to be imported. There may be several queues with different priorities.
```

Author: fixed

  * **ImportReplyQueue** - contains reply message on each import message.

>   * **ImportFailureQueue** - contains import messages that has failed.

Reviewer: ```suggestion
  * **ImportFailureQueue** - contains import messages that have failed.
```

Author: fixed

  * **ExportQueue** - contains exported messages. There may be several queues related to different receivers.
  * **ExportReplyQueue** - contains reply messages on exported messages.
  * **ExportFailureQueue** - contains messages that failed during export.
  * **TriggerQueue** - contains messages that trigger exports.

> * **Activation Request** to/from Scada Gateway and Mesh Data Transfer:

Reviewer: ```suggestion
* **Activation Request** to/from SCADA Gateway and Mesh Data Transfer:
```

Author: fixed

>   * **TimeSeries2ScadaQueue** - contains activation and heartbeat messages to Scada Gateway/Scada.

Reviewer: ```suggestion
  * **TimeSeries2ScadaQueue** - contains activation and heartbeat messages to SCADA Gateway/SCADA.
```

Author: fixed

>   * **AcknowledgeQueue** - contains acknowledgement messages from Scada/Scada Gateway.

Reviewer: ```suggestion
  * **AcknowledgeQueue** - contains acknowledgement messages from SCADA/SCADA Gateway.
```

Author: fixed
